lib/improver/utilities/cube_checker.py

```diff
@@ -140,8 +140,9 @@ def check_cube_coordinates(cube, new_cube, exception_coordinates=None):
         exception_coordinates = []
 
     # Promote available and relevant scalar coordinates
+    cube_dim_names = [coord.name() for coord in cube.dim_coords]
     for coord in new_cube.aux_coords[::-1]:
-        if coord in cube.dim_coords:
+        if coord.name() in cube_dim_names:
```

Comment:
Why has this change been made?

Reply:
This change is to ensure that auxiliary coordinates within the new cube e.g. the orographic enhancement cube are corrected to be dimension coordinates, if the coordinate in question was a dimension coordinate in the old cube e.g. the input precipitation field. This change just checks for the name of the coordinate within the old cube, rather than trying to match the whole coordinate i.e. including the coordinate points and bounds, which may not match.